### ISSUE TITLE
Behind upstream commits/cake build errors

### ISSUE THREAD

**Issue by runvnc** [cozy-labs/emails#2]
Hello I am wondering if the plan is to merge in commits from upstream, or if that is kind of a surgery since this is the light version.

Also I am getting cake build errors and wondering if they are already fixed in upstream.  First problem was missing two spaces in `_menu.styl` and next problem is missing client/public dir.

Thanks.

**Comment by runvnc** [comment]
So far it was just those two minor issues with the build, but still I wonder how compatible this is with upstream?  I am guessing I can't blindly merge it in (236 commits or something).

**Comment by frankrousseau** [comment]
In fact it should be easy. But we changed our database driver. The new one requires some fixes. I will have a look this week and give you more information soon.

@aenario cozydb is not ready for being used with pouchdb some helpers functions use the Data System driver.

**Comment by runvnc** [comment]
Ok so now uses cozydb then?  BTW is this fork mainly for cozy-light and just has a few differences related to that, or what. I actually thought this one was newer but I guess not.  But for cozy-light I need to use this one right?

You didn't also change the way plugins work in emails did you?  I have a whole plan for multi-user set up based on the way it is in this project.

**Comment by frankrousseau** [comment]
In the current version, we hided the plugin system to focus on the main feature. But you can put it back easily I think by changing some lines of code (@clochix any idea is welcome :)).

You're right, this version is for Cozy Light and is older than the current one.

**Comment by runvnc** [comment]
Ok well, I still see config.coffee plugins.  I hope you are not getting rid of cozy-light plugins or americano plugins,they are great and we are using them :)

**Comment by frankrousseau** [comment]
Aaaa ok sorry. I didn't understand you were talking about Americano plugins. Of course they stay!

**Comment by aenario** [comment]
@runvnc, one of the purpose of cozydb was to replace both americano-cozy & americano-cozy-pouchdb. So we could support both cozy & cozy-light with the same codebase.

As frank mentionned, this is not feature complete yet. Once this (and probably a few other bugs) is fixed, the http://github.com/cozy/cozy-emails/ will be the only up-to-date repo and this fork wont be necessary. 

Note that we work on cozy-light in our spare time, so no promise as for an ETA. But if you want to have a go at https://github.com/cozy/cozy-db/issues/15, I will provide assistance & review.

**Comment by runvnc** [comment]
Ok, thanks for explaining.  I am under the impression that we kind of need cozy-light for our project rather than the non-light version (unless we are going to select a totally different webmail/contacts system) so @aenario, I will work on cozy-db 15 today and see if I can help a little.

**Comment by frankrousseau** [comment]
I finally managed to publish a working version of emails! Thank you @runvnc for your patience and congrats @aenario for the ODM and the fixes.

The publishing process will require some improvements, but what matters the most is that it works!

**Comment by runvnc** [comment]
Awesome great job and thanks @frankrousseau and @aenario.  So by publish you mean there is a cozy-emails module in npm or..?  Sorry if that is a dumb question.

**Comment by runvnc** [comment]
I should use this project for now still right instead of upstream.

**Comment by frankrousseau** [comment]
Yes you can do:

```
npm install emails - g
```

Then: 

```
emails
```

**Comment by frankrousseau** [comment]
I just figure out that there is still a problem with Cozy Light. The app is too slow to install.

**Comment by runvnc** [comment]
Well, if it is installing pouchdb, the leveldb compile takes a longtime.  Maybe that is the issue.

**Comment by frankrousseau** [comment]
You're right. This is the issue. I need to extends the install timeout but I don't know which option is required:
https://github.com/frankrousseau/cozy-light/blob/master/lib/helpers/npm.js#L27-L32

**Comment by runvnc** [comment]
https://docs.npmjs.com/misc/config don't see the option

**Comment by runvnc** [comment]
Are you sure npm timed out and not the build script?

**Comment by runvnc** [comment]
I wonder if there could be a version of pouchdb in npm that is built for x86-64 and arm before being published.

**Comment by frankrousseau** [comment]
I'm too tired tonight to figure out what's wrong. I will check later, tomorrow or on monday.

**Comment by runvnc** [comment]
Ok. Get some rest.  I went and tried again on my Linode `npm install pouchdb` the build actually went, quickly, maybe 30 seconds.  This is a 2gb Linode.  I also tried `npm install emails` it went pretty quick and had no problem.

I hope you are not on one of those tiny/micro EC2 instanceS.  They are a scam and will basically shut down under load.  Like it throttles the CPU until useless.

The pouchdb/leveldown install/build on ARM is a lot slower than my Linode, but it did work before on the Beaglebone black.
